Source/JavaScriptCore/runtime/JSGenericTypedArrayViewInlines.h

```diff
@@ -242,14 +242,17 @@ bool JSGenericTypedArrayView<Adaptor>::setWithSpecificType(
     }
     
     // Fail: we need an intermediate transfer buffer (i.e. case (3)).
-    Vector<typename Adaptor::Type, 32> transferBuffer(length);
-    for (size_t i = length; i--;) {
-        transferBuffer[i] = OtherAdaptor::template convertTo<Adaptor>(
-            other->getIndexQuicklyAsNativeValue(i + otherOffset));
+    {
+        Checked<size_t, CrashOnOverflow> sizeToAllocate = length * sizeof(typename Adaptor::Type);
```

Comment:
Wrap length first with Checked. This is wrapping after overflow happens.

Reply:
We should also have a fast path for small size. Previously Vector was allocating 32 and having a fast path for small move. We need to keep it, and going to this path only when it exceeds a certain size.